src/OpenColorIO/fileformats/xmlutils/XMLReaderUtils.h

```diff
@@ -156,16 +158,17 @@ void ParseNumber(const char * str, size_t startPos, size_t endPos, T & value)
     const char * startParse = str + startPos;
 
     double val = 0.0f;
-    char * endParse = nullptr;
-
-    // The strtod expects a C string and str might not be null terminated.
-    // However since strtod will stop parsing when it encounters characters
-    // that it cannot convert to a number, in practice it does not need to
-    // be null terminated.
-    // C++11 version of strtod processes NAN & INF ASCII values.
-    val = strtod(startParse, &endParse);
+
+    size_t adjustedStartPos = startPos;
+    size_t adjustedEndPos = endPos;
+
+    FindSubString(startParse, endPos - startPos, adjustedStartPos, adjustedEndPos);
+
+    const auto result = NumberUtils::from_chars(startParse + adjustedStartPos, startParse + adjustedEndPos, val);
+
     value = (T)val;
-    if (endParse == startParse)
+
+    if (result.ec == std::errc::invalid_argument)
```

Comment:
[Question] Why not using `result.ec != std::errc()` like elsewhere?

Reply:
Because of the semantics of strtod, I signal a parsing overrun by returning `std::errc::argument_out_of_domain`. This is checked by https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/3995e744f5b6e18047129dffdbd5e11794941c5c/tests/cpu/fileformats/xmlutils/XMLReaderUtils_tests.cpp#L60-L62